sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/generator/AbstractQueryGenerator.java

```diff
@@ -64,7 +65,13 @@ private String generateBinaryQuery(@NonNull Criteria criteria, @NonNull List<Pai
         parameters.add(Pair.of(parameter, subjectValue));
 
         if (CriteriaType.isFunction(criteria.getType())) {
-            return getFunctionCondition(ignoreCase, sqlKeyword, subject, parameter);
+            return getFunctionCondition(ignoreCase, sqlKeyword, subject, parameter,
+                CriteriaType.isFunctionWithCaseSensitiveSupport(criteria.getType()));
+        } else if(criteria.getType() == CriteriaType.IS_EQUAL &&
+                ignoreCase != Part.IgnoreCaseType.NEVER &&
+                !Pattern.compile("-?\\d+(\\.\\d+)?").matcher(parameter).matches()) {
```

Comment:
Three things:
- I think you need to check subjectValue not parameter - right now you are just checking whether the parameter name follows the regex?
- For efficiency reason you might want to declare a static final field with the compiled Regex - to avoid compiling it every time 
- the fact that the parameter name vs. actual value didn't fail any of the tests means there is a gap. For integration tests I like your tests. But I think it is appropriate to add unit tests for just this method - to assert that UPPER is only used in the output when appropriate etc. Easiest way to get protection from future regressions that could functionally still be correct (like right now) - but incorrectly fall back to the inefficient path.

Reply:
1. Fixed
2. Since using subjectValue, it is an object so achieved without regex
3. Added tests